src/core/permissions/sagas/getDataSetPermissionsPage.js

```diff
@@ -0,0 +1,198 @@
+import { getOrgDataSetColumnsFromMeta, getOrgDataSetsFromMeta } from '../../edm/actions';
+import { FQNS } from '../../edm/constants';
+import { getOrgDataSetColumnsFromMetaWorker, getOrgDataSetsFromMetaWorker } from '../../edm/sagas';
+import { PAGE_PERMISSIONS_BY_DATA_SET, TOTAL_PERMISSIONS } from '../../redux/constants';
+import {
+  selectOrgDataSets,
+  selectOrgDataSetsColumns,
+  selectOrganization,
+  selectPrincipalPermissions,
+  selectPropertyTypes,
+} from '../../redux/selectors';
+import { GET_DATA_SET_PERMISSIONS_PAGE, getDataSetPermissionsPage, getPermissions } from '../actions';
+
+const { FQN } = Models;
+const { toSagaError } = AxiosUtils;
+const { getPropertyValue } = DataUtils;
+const { isNonEmptyArray, isNonEmptyString } = LangUtils;
+
+const REQUIRED_PROPERTY_TYPES :FQN[] = [
+  FQNS.OL_DATA_SET_ID,
+];
+
+const LOG = new Logger('PermissionsSagas');
+
+function* getDataSetPermissionsPageWorker(action :SequenceAction) :Saga<void> {
+
+  try {
+    yield put(getDataSetPermissionsPage.request(action.id, action.value));
+
+    const {
+      filterByPermissionTypes,
+      filterByQuery,
+      initialize,
+      organizationId,
+      pageSize,
+      principal,
+      start,
+    } :{|
+      filterByPermissionTypes :Array<PermissionType>;
+      filterByQuery :string;
+      initialize :boolean;
+      organizationId :UUID;
+      pageSize :number;
+      principal :Principal;
+      start :number;
+    |} = action.value;
+    let response :WorkerResponse;
+
+    const organization :?Organization = yield select(selectOrganization(organizationId));
+    if (!organization) {
+      throw new Error(ERR_MISSING_ORG);
+    }
+
+    const propertyTypes :Map<UUID, PropertyType> = yield select(selectPropertyTypes(REQUIRED_PROPERTY_TYPES));
+    const propertyTypeIds :Map<FQN, UUID> = propertyTypes.map((propertyType) => propertyType.type).flip();
+    if (propertyTypeIds.count() !== REQUIRED_PROPERTY_TYPES.length) {
+      throw new Error(ERR_MISSING_PROPERTY_TYPE);
+    }
+
+    if (initialize === true) {
+      // NOTE: we only want to call getOrgDataSetsFromMeta once, not for every page
+      // NOTE: we don't need response.data because the redux store will be populated
+      response = yield call(getOrgDataSetsFromMetaWorker, getOrgDataSetsFromMeta({ organizationId }));
+      if (response.error) throw response.error;
+    }
+
+    const dataSets :Map<UUID, Map> = yield select(selectOrgDataSets(organizationId));
+    const dataSetKeys :List<List<UUID>> = dataSets.keySeq().map((dataSetId :UUID) => List([dataSetId])).toList();
+
+    if (initialize === true) {
+      // NOTE: we only want to call getPermissions once, not for every page
+      // NOTE: we don't need response.data because the redux store will be populated
+      response = yield call(getPermissionsWorker, getPermissions(dataSetKeys));
+      if (response.error) throw response.error;
+    }
+
```

Comment:
I think this `if block` and the previous should be coalesced for readability

That is:
```
const dataSets :Map<UUID, Map> = yield select(selectOrgDataSets(organizationId));
const dataSetKeys :List<List<UUID>> = dataSets.keySeq().map((dataSetId :UUID) => List([dataSetId])).toList();

if (initialize) {
  response = yield call(getOrgDataSetsFromMetaWorker, getOrgDataSetsFromMeta({ organizationId }));
  if (response.error) throw response.error;

  response = yield call(getPermissionsWorker, getPermissions(dataSetKeys));
  if (response.error) throw response.error;
}
```

Reply:
I don't think this will work. `getOrgDataSetsFromMeta` is the first thing that needs to happen in order for `selectOrgDataSets` to actually get things, after which `getPermissions` can happen.

If I were to combine in one `if` block, I would have to do `selectOrgDataSets` twice.

Reply:
I don't understand. Are you saying that `selectOrgDataSets` depends on `getOrgDataSetsFromMeta`?

Reply:
Yes.